lib/ansible/modules/monitoring/spectrum_device.py

```diff
@@ -0,0 +1,329 @@
+        model_handle=model_handle,
+        address=model_address,
+        landscape=model_landscape)
+
+    return device
+
+
+def add_device():
+    device_ip = get_ip()
+    device = get_device(device_ip)
+
+    if device:
+        module.exit_json(changed=False, device=device)
+
+    if module.check_mode:
+        device = dict(
+            model_handle=None,
+            address=device_ip,
+            landscape="0x%x" % int(module.params.get('landscape'), 16))
+        module.exit_json(changed=True, device=device)
+
+    resource = 'model?ipaddress=' + device_ip + '&commstring=' + module.params.get('community')
+    resource += '&landscapeid=' + module.params.get('landscape')
+
+    if module.params.get('agentport', None):
+        resource += '&agentport=' + str(module.params.get('agentport', 161))
+
+    result = post(resource)
+    root = ET.fromstring(result)
+
+    if root.get('error') != 'Success':
+        module.fail_json(msg=root.get('error-message'))
+
+    namespace = dict(ca='http://www.ca.com/spectrum/restful/schema/response')
+    model = root.find('ca:model', namespace)
+
+    model_handle = model.get('mh')
+    model_landscape = "0x%x" % int(int(model_handle, 16) / 0x100000 * 0x100000)
+
+    device = dict(
+        model_handle=model_handle,
+        address=device_ip,
+        landscape=model_landscape,
+    )
+
+    module.exit_json(changed=True, device=device)
+
+
+def remove_device():
+    device_ip = get_ip()
+    device = get_device(device_ip)
+
+    if device is None:
+        module.exit_json(changed=False)
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    resource = '/model/' + device['model_handle']
+    result = delete(resource)
+
+    root = ET.fromstring(result)
+
+    namespace = dict(ca='http://www.ca.com/spectrum/restful/schema/response')
+    error = root.find('ca:error', namespace).text
+
+    if error != 'Success':
+        error_message = root.find('ca:error-message', namespace).text
+        module.fail_json(msg="%s %s" % (error, error_message))
+
+    module.exit_json(changed=True)
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec=dict(
+            device=dict(required=True, aliases=['host', 'name']),
+            landscape=dict(required=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+            community=dict(required=True, no_log=True),
+            agentport=dict(type='int', default=161),
+            url=dict(required=True, aliases=['oneclick_url']),
+            url_username=dict(required=True, aliases=['oneclick_user']),
+            url_password=dict(required=True, no_log=True, aliases=['oneclick_password']),
+            use_proxy=dict(type='bool', default='yes'),
+            validate_certs=dict(type='bool', default='yes'),
+        ),
+        required_if=[('state', 'present', ['community'])],
+        supports_check_mode=True
+    )
+
+    if module.params.get('state') == 'present':
```

Comment:
In the `DOCUMENTATION` of the module it does not list `state` as a `required = True` but the module doesn't appear to do anything without that option passed. What's the expected functionality if that option is not provided?

Reply:
If the state parameter is not provide it will assume the default value of 'present' as stated in the doc.